### PR TITLE
Generalized nanopb generator script

### PR DESCRIPTION
**Pull request by dgquintas**
`tools/codegen/core/gen_load_balancing_proto.sh → tools/codegen/core/gen_nano_proto.sh` and generalized the generation of the nanopb output.

Append to `tools/distrib/check_nanopb_output.sh` following the LB example to add checks for new .proto files.

### CODE REVIEW

**Comment by dgquintas** [comment]
ping :)

File: tools/codegen/core/gen_nano_proto.sh [hunk 1]
```diff
@@ -75,34 +76,37 @@ COPYRIGHT_FILE=$(mktemp)
 echo "${COPYRIGHT/<YEAR>/$CURRENT_YEAR}" > $COPYRIGHT_FILE
 
 set -ex
-if [ $# -eq 0 ]; then
-  echo "Usage: $0 <load_balancer.proto> [output dir]"
+if [ $# -lt 2 ]; then
```
Comment (a-veitch) [inline tools/codegen/core/gen_nano_proto.sh]: Or -gt 3 ?

File: tools/codegen/core/gen_nano_proto.sh [hunk 2]
```diff
@@ -75,34 +76,37 @@ COPYRIGHT_FILE=$(mktemp)
...
   exit 3
 fi
 if [[ ! -f "${EXPECTED_OPTIONS_FILE_PATH}" ]]; then
   echo "Expected nanopb options file '${EXPECTED_OPTIONS_FILE_PATH}' missing"
   exit 4
 fi
 
+if [[ "${OUTPUT_DIR:0:1}" != '/' ]]; then
+  echo "The output directory must be an absolute path. Got '$OUTPUT_DIR'"
+  exit 5
```
Comment (a-veitch) [inline tools/codegen/core/gen_nano_proto.sh]: Consider adding a couple of lines documenting exit return values.

**Comment by a-veitch** [comment]
LGTM, couple of minor comments. Thanks for doing this!

File: tools/codegen/core/gen_nano_proto.sh (hunk 2 above)
Comment (dgquintas) [inline tools/codegen/core/gen_nano_proto.sh]: Done.

File: tools/codegen/core/gen_nano_proto.sh (hunk 1 above)
Comment (dgquintas) [inline tools/codegen/core/gen_nano_proto.sh]: Indeed. Done.

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* One or more commits are not associated with a GitHub user. The email used in the Git commit must match a verified email address in a GitHub user profile. For more information, please see https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

_Regards,
CLA GitHub bot_

**Pull request merged**